Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -1520,13 +1520,27 @@ signal_t MatrixWorkspace::getSignalAtCoord(
   }
 
   if (wi < nhist) {
-    const auto &X = this->binEdges(wi);
-    auto it = std::lower_bound(X.cbegin(), X.cend(), x);
-    if (it == X.end()) {
+    const auto &xVals = x(wi);
+    auto it = std::lower_bound(xVals.cbegin(), xVals.cend(), xCoord);
+    if (it == xVals.end()) {
+      if (!isHistogramData()) {
+        auto binEnd =
```

Comment:
I just had another think about this. If the value is not found couldn't it also be in a "bin" before the first point?

Reply:
No, I don't think so. `std::lower_bound` returns the first point that is greater than or equal to `xCoord`.

Reply:
Good point. My brain was misfiring for a minute there.